src/main/scala/beam/agentsim/infrastructure/ChargingNetworkManager.scala

```diff
@@ -59,7 +60,7 @@ class ChargingNetworkManager(
     beamServices.matsimServices.getScenario.getPopulation.getPersons.asScala.map { case (personId, person) =>
       personId -> (person.getSelectedPlan.getPlanElements.asScala
         .find(_.isInstanceOf[Activity])
-        .map(_.asInstanceOf[Activity].getEndTime)
+        .flatMap(x => x.asInstanceOf[Activity].getEndTime.toOption)
```

Comment:
This another place of possible change (where an empty becomes None instead of Double Negative Infinity)